cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/SourceControlManagementHttpHandlerTests.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.services.http.handlers;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import io.cdap.cdap.gateway.handlers.SourceControlManagementHttpHandler;
+import io.cdap.cdap.internal.app.services.http.AppFabricTestBase;
+import io.cdap.cdap.proto.sourcecontrol.AuthType;
+import io.cdap.cdap.proto.sourcecontrol.Provider;
+import io.cdap.cdap.proto.sourcecontrol.RepositoryConfig;
+import io.cdap.common.http.HttpResponse;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.lang.reflect.Type;
+
+/**
+ * Tests for {@link SourceControlManagementHttpHandler}
+ */
+public class SourceControlManagementHttpHandlerTests extends AppFabricTestBase {
+  
+  private static final String NAME = "test";
+  private static final Gson GSON = new Gson();
+
+  private void assertResponseCode(int expected, HttpResponse response) {
+    Assert.assertEquals(expected, response.getResponseCode());
+  }
+
+  private RepositoryConfig readGetRepositoryResponse(HttpResponse response) {
+    Type typeToken = new TypeToken<RepositoryConfig>() { }.getType();
```

Comment:
don't need to do this, can just use `RepositoryConfig.class`. TypeToken is used when there are generics, like if it was going to be a `List<RepositoryConfig>`